BUG: Ensure consistent interpretation of uint64 states.

rkern opened the pull request:
This would have been a problem on big-endian machines. We don't seem to have any in our CI fleet. Do we have any volunteers with a Power machine laying around?

rkern committed:
BUG: Ensure consistent interpretation of uint64 states.

rkern committed:
BUG: make sure all the bytes are in the right order, too.

charris commented:
I believe @ahaldane has access to a big-endian machine.

rkern committed:
TST: explicitly test uint64 generation.

charris commented:
If you do merge another branch into this, a simple rebase will straighten out the git graph.

rkern committed:
BUG: ensure that Generator.bytes() is consistent on all platforms.

rkern committed:
BUG: Make RandomState.bytes() big-endian-safe.

This is a new bug that was introduced with the new RandomState implementation.
The old code was already doing this.

mattip commented:
Confirmed that d8a1914 passes all tests on a big-endian machine.
Thanks @rkern
